main.tf

```diff
@@ -19,6 +19,7 @@ locals {
 }
 
 resource "google_pubsub_topic" "topic" {
+  count   = var.topic == "" ? 0 : 1
```

Comment:
As you noted, this will cause an issue if `topic` is a computed property.

Instead, could we add an explicit `create_topic` variable?